tools/sigma/backends/logiq.py

```diff
@@ -0,0 +1,90 @@
+import re
+from .base import BaseBackend
+from .mixins import QuoteCharMixin
+import json
+
+class LogiqBackend(BaseBackend, QuoteCharMixin):
+    """Generates Perl compatible regular expressions and puts 'grep -P' around it"""
+    identifier = "logiq"
+    active = True
+    config_required = False
+
+    reEscape = re.compile("([\\|()\[\]{}.^$+])")
+
+    def generate(self, sigmaparser):
+        """Method is called for each sigma rule and receives the parsed rule (SigmaParser)"""
+
+        eventRule = dict()
+        eventRule["name"] = sigmaparser.parsedyaml["title"]
+        eventRule["groupName"] = sigmaparser.parsedyaml["logsource"]["product"]
+        eventRule["description"] = sigmaparser.parsedyaml["description"]
+        eventRule["condition"] = sigmaparser.parsedyaml["detection"]
+        eventRule["level"] = sigmaparser.parsedyaml["level"]
+
+        # for key,value in eventRule.items():
+        #     print(key, ":", value)
+
+        for parsed in sigmaparser.condparsed:
+            query = self.generateQuery(parsed)
+            before = self.generateBefore(parsed)
+            after = self.generateAfter(parsed)
+
+            eventRule["condition"] = ""
+            if before is not None:
+                eventRule["condition"] = before
+            if query is not None:
+                eventRule["condition"] += query
+            if after is not None:
+                eventRule["condition"] += after
+
+            return json.dumps(eventRule)
+
+    def generateQuery(self, parsed):
+        # print("generateQuery: ", parsed)
+        return "%s" % self.generateNode(parsed.parsedSearch)
+
+    def cleanValue(self, val):
+        if val[0] == '*':
+            val = val.replace("*","/*")
+      
+        # print("cleanValue: ", val)
+        return val
+
+    def generateORNode(self, node):
+        # print("generateORNode: ", node)
+        return "%s" % " || ".join([self.generateNode(val) for val in node])
+
+    def generateANDNode(self, node):
+        # print("generateORNode: ", node)
+        return "%s" % " && ".join([self.generateNode(val) for val in node])
+        
+    def generateNOTNode(self, node):
+        # print("generateNOTNode: ", node)        
+        return "%s" % self.generateNode(node.item)
```

Comment:
This passes the value as-is, is this correct? Would expect a negation operator here.